### PR TITLE
feat: javascript modules auto deployment to npm based on labels

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

Adds automation for javascript packages do be published and released based on pull request labels.
It also automatically adds the pending changelog entries (everything under the [TBD] title) to the github releases and updates the changelog file with a new [TBD] section.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.

### CODE REVIEW

File: .github/workflows/javascript_publish.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        with:
+          prerelease: false
+          body_path: ${{ steps.notes.outputs.notes }}
+      - name: Update Changelog
+        run: |
+          git fetch
+          git pull
+          git config --global user.name "Arabot-1"
+          git config --global user.email "arabot-1@users.noreply.github.com"
+          sed -i "s/\[TBD\]/[TBD]  \n  \n\## [1.0.5\] - $(date)/" "javascript/modules/${{ needs.checkTag.outputs.package }}/CHANGELOG.md"
```
Comment (brickpop) [inline .github/workflows/javascript_publish.yml]: Is the `1.0.5` literal expected?

Comment (mathewmeconry) [reply .github/workflows/javascript_publish.yml]: uhh nop. good spot. I'll change that shortly

**Review by marcvelmer — COMMENTED** [review]
Great job!

I think this should immediately be merged once we have the unit testing working on the CI, which will be the first issue to be tackled on this new sprint.

**Comment by brickpop** [comment]
Thank you for the changes @mathewmeconry!

I wonder if there is a way to start/stop the ganache service which is self contained in the tests. In dvote-js we were doing something like this programmatically:
- https://github.com/vocdoni/dvote-js/blob/main/packages/client/test/helpers/web3-service.ts
- https://github.com/vocdoni/dvote-js/blob/main/packages/client/test/integration/processes.ts#L54-L70

I thought that there would be a way to do this with HardHat, but can't seem to find anything in the docs :-/

Similar thing with the contract deployment, is it doable to deploy them from the test itself within the `beforeEach()`?

**Comment by mathewmeconry** [comment]
> Thank you for the changes @mathewmeconry!
> 
> I wonder if there is a way to start/stop the ganache service which is self contained in the tests. In dvote-js we were doing something like this programmatically:
> 
> * https://github.com/vocdoni/dvote-js/blob/main/packages/client/test/helpers/web3-service.ts
> * https://github.com/vocdoni/dvote-js/blob/main/packages/client/test/integration/processes.ts#L54-L70
> 
> I thought that there would be a way to do this with HardHat, but can't seem to find anything in the docs :-/
> 
> Similar thing with the contract deployment, is it doable to deploy them from the test itself within the `beforeEach()`?

done. I added it into the beforeAll because then we can speed up the tests a bit and the tests don't need a clean factory every time.

File: javascript/modules/client/src/context.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { UnsupportedProtocolError } from "@aragon/sdk-common";
 export { ContextParams } from "./internal/interfaces/context";
 
 const supportedProtocols = ["https:"];
-if(process.env.CI) {
+if(process.env.TESTING) {
```
Comment (brickpop) [inline javascript/modules/client/src/context.ts]: I see this was already introduced, but I'm not sure if the NodeJS-only `process.env` will bring us trouble when running on the browser

**Comment by brickpop** [comment]
The PR looks perfect, my only doubt is about introducing the `process` dependent piece of code

File: javascript/modules/client/src/context.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { UnsupportedProtocolError } from "@aragon/sdk-common";
 export { ContextParams } from "./internal/interfaces/context";
 
 const supportedProtocols = ["https:"];
-if(process.env.TESTING) {
+if(process && process.env && process.env.TESTING) {
```
Comment (brickpop) [inline javascript/modules/client/src/context.ts]:
```suggestion
if(typeof process !== "undefined" && process.env?.TESTING) {
```

Comment (mathewmeconry) [reply javascript/modules/client/src/context.ts]: The change you suggest only checks for undefined. if you don't use typeof it also checks for empty strings, 0 and NULL

**Review by brickpop — APPROVED** [review]

**Review by marcvelmer — APPROVED** [review]

**Pull request merged**